core/orchestrator.go

```diff
@@ -102,26 +102,43 @@ func (orch *orchestrator) ProcessPayment(payment net.Payment, manifestID Manifes
 		return nil
 	}
 
-	ticket := &pm.Ticket{
-		Recipient:         ethcommon.BytesToAddress(payment.Ticket.Recipient),
-		Sender:            ethcommon.BytesToAddress(payment.Ticket.Sender),
-		FaceValue:         new(big.Int).SetBytes(payment.Ticket.FaceValue),
-		WinProb:           new(big.Int).SetBytes(payment.Ticket.WinProb),
-		SenderNonce:       payment.Ticket.SenderNonce,
-		RecipientRandHash: ethcommon.BytesToHash(payment.Ticket.RecipientRandHash),
+	if payment.TicketParams == nil {
+		return errors.Wrapf(errors.New("No TicketParams in payment"), "Could not find TicketParams for payment %v", payment)
 	}
-	seed := new(big.Int).SetBytes(payment.Seed)
 
-	sessionID, won, err := orch.node.Recipient.ReceiveTicket(ticket, payment.Sig, seed)
-	if err != nil {
-		return errors.Wrapf(err, "error receiving ticket for payment %v for manifest %v", payment, manifestID)
+	if payment.Sender == nil || len(payment.Sender) == 0 {
+		return errors.Wrapf(errors.New("No sender in payment"), "Payment %v does not contain a sender", payment)
 	}
 
-	if won {
-		glog.V(common.DEBUG).Info("Received winning ticket")
-		cachePMSessionID(orch.node, manifestID, sessionID)
+	seed := new(big.Int).SetBytes(payment.TicketParams.Seed)
+
+	ticket := &pm.Ticket{
+		Recipient:         ethcommon.BytesToAddress(payment.TicketParams.Recipient),
+		Sender:            ethcommon.BytesToAddress(payment.Sender),
+		FaceValue:         new(big.Int).SetBytes(payment.TicketParams.FaceValue),
+		WinProb:           new(big.Int).SetBytes(payment.TicketParams.WinProb),
+		RecipientRandHash: ethcommon.BytesToHash(payment.TicketParams.RecipientRandHash),
 	}
 
+	for _, tsp := range payment.TicketSenderParams {
+
+		ticket.SenderNonce = tsp.SenderNonce
+
+		sessionID, won, err := orch.node.Recipient.ReceiveTicket(
+			ticket,
+			tsp.Sig,
+			seed,
+		)
+		if err != nil {
+			err = errors.Wrapf(err, "Error receiving ticket %v for manifest %v", ticket, manifestID)
```

Comment:
Since we are just logging `err` here, we can log the error as-is instead of wrapping it with another error message since we are not propagating it to a caller. `errors.Wrapf` is primarily useful for annotating an error so that you can maintain an error stack trace as an error is propagated up the caller stack.

Reply:
Considering that `ReceiveTicket` errors don't return useful information for which ticket errored in a multi-ticket setup, it might be useful to at least add that information to the error message through formatting? 

eg. 

`glog.Errorf("Error receiving ticket %v for manifest %v: %v\n", ticket, manifestID, err)`

running into an invalid `WinProb` would log 

`Error receiving ticket <ticket> for manifest <manifest>: invalid ticket winProb <winProb>`

Reply:
Yeah we could include the extra information in the logged error message as you describe in the first `glog` code snippet you listed.

I slightly misspoke in my original comment - I intended to say we don't need to use `errors.Wrap` here to wrap `err` i.e. we don't need L133 which modifies `err` with `errors.Wrapf`. However, I *do* think it is useful to add extra contextual information when logging `err` with `glog` via a formatted error message.